File: VB/DXSample/Data/SimpleData.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Collections.Generic

Namespace DXSample
	Public Class SimpleData
		Inherits SimpleDataBase

		Private _Bool As Boolean
		Private _Name As String
		Private _ID As Integer

		Public Property ID() As Integer
			Get
				Return _ID
			End Get
			Set(ByVal value As Integer)
				If _ID = value Then
					Return
				End If
				_ID = value
                OnPropertyChanged("ID")
			End Set
		End Property

		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				If _Name = value Then
					Return
				End If
				_Name = value
                OnPropertyChanged("Name")
			End Set
		End Property

		Public Property Bool() As Boolean
			Get
				Return _Bool
			End Get
			Set(ByVal value As Boolean)
				If _Bool = value Then
					Return
				End If
				_Bool = value
                OnPropertyChanged("Bool")
			End Set
		End Property
	End Class
End Namespace

File: CS/DXSample/Data/SimpleData.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace DXSample {
    public class SimpleData : SimpleDataBase {

        private bool _Bool;
        private string _Name;
        private int _ID;

        public int ID {
            get { return _ID; }
            set {
                if (_ID == value)
                    return;
                _ID = value;
                OnPropertyChanged(GetPropertyName(() => ID));
            }
        }

        public string Name {
            get { return _Name; }
            set {
                if (_Name == value)
                    return;
                _Name = value;
                OnPropertyChanged(GetPropertyName(() => Name));
            }
        }

        public bool Bool {
            get { return _Bool; }
            set {
                if (_Bool == value)
                    return;
                _Bool = value;
                OnPropertyChanged(GetPropertyName(() => Bool));
            }
        }
    }
}
